core/foundation/src/TClassEdit.cxx

```diff
@@ -1533,14 +1537,17 @@ static void ResolveTypedefImpl(const char *tname,
       }
       while (tname[cursor]==' ') ++cursor;
    }
-
+   if (tname[cursor]=='c' && (cursor+6<len)) {
+      if (strncmp(tname+cursor,"class ",6) == 0) {
+         cursor += 6;
+      }
+   }
    if (tname[cursor]=='c' && (cursor+6<len)) {
```

Comment:
BTW, you could simplify it like this:
```suggestion
```

Reply:
AFAICS, in your commit there is twice the same if statement so the `constprefix ` will be set in the second one anyway. Or did I overlook your commit?

Reply:
Mm it's almost the same, but in the second one the index cursor is cursor+6, and it's str-comparing against class instead of const, so we are not doing exactly the same.
Class int will go into the first if but not into the second since cursor has increased so constprefix will be false

Reply:
Well, as you prefer, it's only a repetition of identical code anyway, I guess the compiler will optimize it

Reply:
Actually, doesn't it depends on whether the name can sometimes have both `const ` and `class ` and in which order?

Reply:
Either way, the code should be augment with a comment expressing that we looked into that and what the answer was.